### PR TITLE
build(deps): downgrade `argoproj/argo-cd` to `v2.8.1`

### PR DESCRIPTION
**Pull request by onematchfox**
TLDR: This PR downgrades `argoproj/argocd` to `v2.8.1` to unblock our releases whilst still :crossed_fingers: ensuring that the functionality added in #337 still works. 

---- 

Our release build is currently [broken](https://github.com/oboukili/terraform-provider-argocd/actions/runs/6920187726). I had hoped that bumping versions (#347) in the GitHub action would fix this, but it turns out that this is due to an issue in the version of `argoproj/argo-cd` that we currently reference. 

Goreleaser log (`v2.8.3`):
```
$ curl -sfL https://goreleaser.com/static/run | bash           
Using the OSS distribution...
Downloading GoReleaser v1.22.1...
Verifying checksums...
Could not verify signatures, cosign is not installed.
  • starting release...
  • loading                                          path=.goreleaser.yml
  • loading environment variables
    • using token from  $GITHUB_TOKEN 
  • getting and validating git state
    • git state                                      commit=18ca584c12adfd161de77a8c62389e204332cf95 branch=master current_tag=v6.1.0 previous_tag=v6.0.3 dirty=false
  • parsing tag
  • setting defaults
  • checking distribution directory
  • loading go mod information
  • build prerequisites
  • writing effective config file
    • writing                                        config=dist/config.yaml
  • building binaries
    • building                                       binary=dist/terraform-provider-argocd_freebsd_arm_6/terraform-provider-argocd_v6.1.0
    • building                                       binary=dist/terraform-provider-argocd_windows_arm_6/terraform-provider-argocd_v6.1.0.exe
    • building                                       binary=dist/terraform-provider-argocd_darwin_amd64_v1/terraform-provider-argocd_v6.1.0
    • building                                       binary=dist/terraform-provider-argocd_linux_amd64_v1/terraform-provider-argocd_v6.1.0
    • building                                       binary=dist/terraform-provider-argocd_freebsd_amd64_v1/terraform-provider-argocd_v6.1.0
    • building                                       binary=dist/terraform-provider-argocd_linux_arm_6/terraform-provider-argocd_v6.1.0
    • building                                       binary=dist/terraform-provider-argocd_linux_arm64/terraform-provider-argocd_v6.1.0
    • building                                       binary=dist/terraform-provider-argocd_darwin_arm64/terraform-provider-argocd_v6.1.0
    • building                                       binary=dist/terraform-provider-argocd_windows_amd64_v1/terraform-provider-argocd_v6.1.0.exe
    • took: 3m53s
  ⨯ release failed after 3m53s               error=failed to build for freebsd_arm_6: exit status 1: # github.com/argoproj/argo-cd/v2/util/http
../../../go/pkg/mod/github.com/argoproj/argo-cd/v2@v2.8.3/util/http/http.go:22:77: cannot use math.MaxInt64 (untyped int constant 9223372036854775807) as int value in argument to env.ParseNumFromEnv (overflows)
```

The bug in `argo-proj/argocd` was introduced in https://github.com/argoproj/argo-cd/pull/14979, which (per [this comment](https://github.com/argoproj/argo-cd/pull/14979#issuecomment-1692272326)) was first released in `v2.8.2`. We currently reference `v2.8.3` which was  introduced as part of #337 (added support for application set sync policies).

Initially, I thought we may as well upgrade to the latest version of `argoproj/argocd` as this would address #326 at the same time. However, unfortunately, the issue still exists with `v2.9.1`. Goreleaser log (`v2.8.3`):

```
...
  ⨯ release failed after 1m43s               error=failed to build for linux_arm_6: exit status 1: # github.com/argoproj/argo-cd/v2/util/http
../../../go/pkg/mod/github.com/argoproj/argo-cd/v2@v2.9.1/util/http/http.go:22:77: cannot use math.MaxInt64 (untyped int constant 9223372036854775807) as int value in argument to env.ParseNumFromEnv (overflows)
# github.com/argoproj/argo-cd/v2/util/settings
../../../go/pkg/mod/github.com/argoproj/argo-cd/v2@v2.9.1/util/settings/settings.go:564:22: undefined: enginecache.RespectRbacDisabled
../../../go/pkg/mod/github.com/argoproj/argo-cd/v2@v2.9.1/util/settings/settings.go:569:23: undefined: enginecache.RespectRbacNormal
../../../go/pkg/mod/github.com/argoproj/argo-cd/v2@v2.9.1/util/settings/settings.go:571:23: undefined: enginecache.RespectRbacStrict
../../../go/pkg/mod/github.com/argoproj/argo-cd/v2@v2.9.1/util/settings/settings.go:573:23: undefined: enginecache.RespectRbacDisabled
../../../go/pkg/mod/github.com/argoproj/argo-cd/v2@v2.9.1/util/settings/settings.go:576:21: undefined: enginecache.RespectRbacDisabled
```

Hence.... Downgrading for now... As soon as I get the chance, I will try to put a fix through to `argoproj/argocd`.

### CODE REVIEW

**Comment by mkilchhofer** [comment]
I am pretty sure that this happens as you also build for ARM 32 bit arch:

https://github.com/oboukili/terraform-provider-argocd/blob/18ca584c12adfd161de77a8c62389e204332cf95/.goreleaser.yml#L14-L19

But yes, I also think that this should be addressed upstream in argoproj/argo-cd 👍
